.github/workflows/ci.yml

```diff
@@ -16,7 +16,6 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
         python-version:
-        - '3.8'
```

Comment:
Okay, so with 3.8 workflow what's happening is a conflict in `numpy`, because we've updated all the versions. 

Something like what's happening here https://github.com/openedx/openedx-calc/actions/runs/8231433530/job/22568231165?pr=99#step:7:11 

So should I downgrade numpy version and check if that'll work with both 3.8 and 3.12 ?

Reply:
Normally you can add a version of numpy that will work with both versions to `https://github.com/openedx/openedx-chem/blob/master/requirements/constraints.txt` along with a comment that we can remove the constraint once the 3.8 testing is removed.

With the numpy package in particular, it may make more sense to add it here: https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt since we use it in many packages across our ecosystem.

Reply:
@feanil current version of numpy with python 3.8 is `1.23.x` here as well as in https://github.com/openedx/openedx-calc/pull/99  and it's being upgraded to `1.26.4`. 

I manually tried to find a common version between them, python 3.8 will support till `numpy==1.24.x` and if we do try to install it with python 3.12, it will give `AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?`  This error comes as a result of 3.12 removing `ImpImporter` [ref](https://docs.python.org/3/whatsnew/3.12.html#importlib:~:text=importlib.abc.Finder%2C%20pkgutil.ImpImporter%2C%20and%20pkgutil.ImpLoader%20have%20been%20removed.%20(Contributed%20by%20Barry%20Warsaw%20in%20gh%2D98040.)) or it could be because numpy depends on old pip.  Also I tried to look into numpy notes but could not find an overlap with a version that'd support 3.8 and 3.12

with python 3.8, numpy maximum version supported is 1.24.4 
and with python 3.12, numpy minimum version supported is 1.26.0 [numpy news](https://numpy.org/news/)

Reply:
Yea, I noticed this as well in a different spot that pulls numpy.  Given the importance of the package, I think we'll have to target 3.11 for edx-platform and its dependencies first and then do the 3.12 upgrade in the future.

Reply:
Hi @feanil  okay, so should I raise another similar PR ? with 3.11 support right ?

Reply:
I think a separate PR to add testing for 3.11 and fixing any bugs that causes makes the most sense I think.